src/main.py

```diff
@@ -168,19 +175,47 @@ def run_plan_bot(
             praw.models.Comment(reddit, _data=pushshift_comment)
         )
 
-        process_post(
-            comment,
-            plans,
-            posts_db,
-            post_ids_processed,
-            send=send_replies,
-            simulate=simulate_replies,
-            skip_tracking=skip_tracking,
-        )
+        process_the_post(comment)
+
+    # Get new comments since we last ran.
+    #
+    # subreddit.comments() returns the newest comments first so we
+    # need to keep track of that and save that as our cursor.  With no
+    # specified params, it returns newest 100 comments in the
+    # subreddit.
+    comments_params = get_comments_params(comments_progress_ref)
+
+    current_newest_comment_id = None
+    for comment in subreddit.comments(params=comments_params):
```

Comment:
If you change this line to `for comment in reversed(list(subreddit.comments(params=comments_params))):` then it'll be sorted by oldest to newest, and then you won't have the issue mentioned below, and I think the behavior will be better in case the bot crashes in the middle of a run!

Reply:
True. That mid-run crash always throws a wrench in the works.